### PR TITLE
Fix JSONL loader handling of blank/invalid lines

### LINKED ISSUE #2424: JSONL reader skips entire file when input contains blank lines
### Describe the issue

`JSONLinesFileReader.read_file()` parses every line returned by `text.splitlines()` with `json.loads(line)`. If a `.jsonl` file contains an accidental blank or whitespace-only line between valid records, `json.loads("")` raises `JSONDecodeError`.

That exception bubbles up to `InputReader._iterate_files()`, where the reader catches the file-level exception, logs a warning, and skips the entire file. As a result, valid records before and after the blank line are not loaded.

### Steps to reproduce

Create a JSONL input file:

```jsonl
{"text":"a"}

{"text":"b"}
```

Load it through the JSONL input reader.

### Expected behavior

The blank or whitespace-only line is ignored and the two valid JSONL records are loaded.

### Actual behavior

The blank line raises `JSONDecodeError`, and the parent input reader skips the whole file.

### Additional context

This is a small parser robustness issue. It does not change handling for malformed non-empty JSONL lines.

A proposed fix is available in #2423.

### PR DESCRIPTION
**Pull request by natoverse**
## Summary
- Skip blank/whitespace-only lines in JSONL input loader
- Skip malformed JSON rows safely
- Skip non-object JSON rows
- Add regression test and fixture coverage
- Add semversioner patch entry

Fixes #2424

### CODE REVIEW

**Comment by natoverse** [comment]
This replaces https://github.com/microsoft/graphrag/pull/2423 to include more handling of edge cases

File: tests/unit/indexing/input/test_jsonl_loader.py [hunk 1]
```diff
@@ -40,3 +40,20 @@ async def test_jsonl_loader_one_file_with_title():
     documents = await reader.read_files()
     assert len(documents) == 3
```
Comment (gaudyb) [inline tests/unit/indexing/input/test_jsonl_loader.py:41]: for this test, is it better to have this fixed value "3" as a default value on top of the test? to understand better why and also to change it in case you want to test something else.

**Review by dworthen — APPROVED** [review]
